### ISSUE TITLE
Constructor signature rule (analogous to function signature rule)

### ISSUE THREAD

**Issue by terminalnode** [pinterest/ktlint#2090]
## Expected Rule behavior
There is an experimental function signature rule which is fantastic. I was first going to file a bug report that this is not applied to constructors, but on closer inspection of the different cases I think perhaps constructor validation should be a different rule.

This config enabled both function signature and the proposed constructor signature:
```yml
[*.{kt,kts}]
ktlint_standard_function-signature = enabled
ktlint_standard_constructor-signature = enabled
ktlint_function_signature_rule_force_multiline_when_parameter_count_greater_or_equal_than = 2
```
They both use the same force multiline rule for consistency.

The most clear-cut case is for secondary constructors:
```kt
// Allowed
data class TestGood(val a: Int) {
  // Parameter count < 2, multiline allowed but not enforced
  constructor(a: String) : this(a.toInt())

  // Parameter count < 2, multiline allowed but not enforced
  constructor(
    a: Long,
  ) : this(a.toInt())

  // Multiline enforced
  constructor(
    a: Int,
    b: Int,
  ) : this(a + b)
}

// Disallowed
data class TestBad(val a: Int) {
  constructor(a: Int, b: Int) : this(a + b)
}
```

Then there is the case of primary constructors. These are up to debate, but IMO they should be treated the same way as secondary constructors.
```kt
// Passes - parameter count < 2, multiline allowed but not enforced
data class TestGood1(val a: Int)

// Passes - parameter count < 2, multiline allowed but not enforced
data class TestGood2(
  val a: Int,
)

// Passes - parameter count ≥ 2, multiline enforced
data class TestGood3(
  val a: Int,
  val b: Int,
)

// Fails - parameter count ≥ 2, multiline enforced
data class TestBad(val a: Int, val b: Int)
```

**Comment by paul-dingemans** [comment]
> There is an experimental function signature rule which is fantastic. I was first going to file a bug report that this is not applied to constructors, but on closer inspection of the different cases I think perhaps constructor validation should be a different rule.

I am very pleased that you like the function-signature rule. It is also one of my favorite rules. Your examples should be taken into account in the scope of #1349 which was already planned for the 1.0 release.

**Comment by paul-dingemans** [comment]
Implemented in #1349

**Comment by paul-dingemans** [comment]
See #2119
